geopandas/tools/clip.py

```diff
@@ -220,38 +151,30 @@ def clip(gdf, clip_obj):
         (geom_types == "LineString")
         | (geom_types == "LinearRing")
         | (geom_types == "Polygon")
+        | (geom_types == "MultiPolygon")
+        | (geom_types == "MultiLineString")
     )
-    multipoly_line_idx = np.asarray(
-        (geom_types == "MultiPolygon") | (geom_types == "MultiLineString")
-    )
-    point_idx = np.asarray((geom_types == "Point"))
-    multipoint_idx = np.asarray((geom_types == "MultiPoint"))
 
-    multipoints = gdf[multipoint_idx]
-    if not multipoints.empty:
-        multipoint_gdf = _clip_multi_point(multipoints, poly)
-    else:
-        multipoint_gdf = None
+    point_idx = np.asarray((geom_types == "Point") | (geom_types == "MultiPoint"))
 
     points = gdf[point_idx]
     if not points.empty:
         point_gdf = _clip_points(points, poly)
     else:
         point_gdf = None
 
-    multipoly_lines = gdf[multipoly_line_idx]
-    if not multipoly_lines.empty:
-        multipoly_line_gdf = _clip_multi_poly_line(multipoly_lines, poly)
-    else:
-        multipoly_line_gdf = None
-
     poly_lines = gdf[poly_line_idx]
     if not poly_lines.empty:
         poly_line_gdf = _clip_line_poly(poly_lines, poly)
     else:
         poly_line_gdf = None
 
     order = pd.Series(range(len(gdf)), index=gdf.index)
-    concat = pd.concat([multipoint_gdf, point_gdf, multipoly_line_gdf, poly_line_gdf])
+    concat = pd.concat([point_gdf, poly_line_gdf])
     concat["_order"] = order
+    if "GeometryCollection" in concat.geom_type[0]:
```

Comment:
@martinfleis i am going to look at this part in a bit more detail. @nkorinek mentioned there might be a better way here but i'm not clear what causes the keyerror yet - it had something to do with not all features having a geom_type.

Reply:
actually i see another issue here not ... `geom_type[0]` is not a good way to test things because what if the second row has a collection.